microsoft/testsuites/core/azure_image_standard.py

```diff
@@ -891,3 +897,74 @@ def verify_no_pre_exist_users(self, node: Node) -> None:
             user_name, user_passwd = passwd_raw_output.split(":")[0:2]
             if not ("*" in user_passwd or "!" in user_passwd or "x" == user_passwd):
                 raise LisaException(f"password of user {user_name} should be deleted.")
+
+    @TestCaseMetadata(
+        description="""
+        This test will check that the readme file existed in resource disk mount point.
+
+        Steps:
+        1. Get the mount point for the resource disk. If `/var/log/cloud-init.log`
+        file is present, mount location is `/mnt`, otherwise it is obtained from
+        `ResourceDisk.MountPoint` entry in `waagent.conf` configuration file.
+        2. Verify that resource disk is mounted from the output of `mount` command.
+        3. Verify lost+found folder exists.
+        4. Verify DATALOSS_WARNING_README.txt file exists.
+        5. Verify 'WARNING: THIS IS A TEMPORARY DISK' contained in
+        DATALOSS_WARNING_README.txt file.
+        """,
+        priority=1,
+        requirement=simple_requirement(
+            disk=AzureDiskOptionSettings(has_resource_disk=True),
+            supported_platform_type=[AZURE],
+        ),
+    )
+    def verify_resource_disk_readme_file(self, log: Logger, node: RemoteNode) -> None:
+        resource_disk_mount_point = get_resource_disk_mount_point(log, node)
+
+        # verify that resource disk is mounted
+        # function returns successfully if disk matching mount point is present
+        node.features[Disk].get_partition_with_mount_point(resource_disk_mount_point)
+
+        # verify lost+found folder exists
+        fold_path = f"{resource_disk_mount_point}/lost+found"
+        folder_exists = node.tools[Ls].path_exists(fold_path, sudo=True)
+        assert_that(folder_exists, f"{fold_path} should be present").is_true()
+
+        # verify DATALOSS_WARNING_README.txt file exists
+        file_path = f"{resource_disk_mount_point}/DATALOSS_WARNING_README.txt"
+        file_exists = node.tools[Ls].path_exists(file_path, sudo=True)
+        assert_that(file_exists, f"{file_path} should be present").is_true()
+
+        # verify 'WARNING: THIS IS A TEMPORARY DISK' contained in
+        # DATALOSS_WARNING_README.txt file.
+        read_text = node.tools[Cat].read(file_path, force_run=True, sudo=True)
+        assert_that(
+            read_text,
+            f"'WARNING: THIS IS A TEMPORARY DISK' should be present in {file_path}",
+        ).contains("WARNING: THIS IS A TEMPORARY DISK")
+
+    @TestCaseMetadata(
+        description="""
+        This test will check that resource disk is formatted correctly.
+
+        Steps:
+        1. Get the mount point for the resource disk. If `/var/log/cloud-init.log`
+        file is present, mount location is `/mnt`, otherwise it is obtained from
+        `ResourceDisk.MountPoint` entry in `waagent.conf` configuration file.
+        2. Verify that resource disk file system type should not be 'ntfs'.
```

Comment:
[nit] Can you please provide reason why this should not be NTFS?

Reply:
by default, the resource disk comes with ntfs, I added this case for one customer incident, there is hung caused by udevadm settle command, then the resource disk is not formatted correctly.

Reply:
updated comments